### ISSUE TITLE
browserify --node with require.extentions

### ISSUE THREAD

**Issue by wesleytodd** [substack/node-browserify#1564]
This seems to not work.  Sorry if I missed another issue that mentions it, but I am using version `12.0.2` and doing a command like this:

```
$ browserify --node -t [babelify] --debug -u tinyliquid [A bunch of other ignores...] bin/app -o build/bin/app
```

One of the modules that it includes is [`consolidate`](https://www.npmjs.com/package/consolidate), which uses `require.extentions` to modify imports.  I am probably going to move away from that package after having these issues and seeing that it registers those hooks whether or not we are using that template engine, but the issue stands that I cannot run my bundle in node if it runs anything that uses that feature.

I looked at https://github.com/substack/browser-pack and see that maybe the issue is in the prelude, that it would need to include proxying the `require.extentions` field, but figured this issue is better opened here for discussion.  Tell me if you would rather this be moved somewhere else (edit: or closed) :)

*Also I understand that bundling the app together means that `require.extentions` is not used, regardless the code doesn't run.*

**Comment by terinjokes** [comment]
As you are likely already aware, [`require.extension`](https://nodejs.org/api/globals.html#globals_require_extensions) is deprecated. Browserify doesn't support it (consider that a subtle bug), as it doesn't make much sense in a browser environment (where you'd likely much rather compile ahead of time).

I don't see us spending any time attempting to add this functionality as a result.

**Comment by wesleytodd** [comment]
Actually @terinjokes I did not know, but I intentionally don't use it because I was aware of it's issues.  Just so happens that a module I DO use uses it in a way that is both unexpected and IMHO bad.  As it turns out I was able to get a more up-to-date version of that module where the hard error was fixed.

Maybe this would be a good small thing I could add to the docs for the `--node` flag?  Something like "This flag does not have feature parity with node's require, for example it does not support the deprecated require.extentions field.  Use at your own risk."?

Thanks for the input and I totally understand the choice to not support it.
